src/modules/dashboard/sagas/actions/managePermissionsMotion.ts

```diff
@@ -0,0 +1,204 @@
+import { hexlify, hexZeroPad } from 'ethers/utils';
+
+import { ContextModule, TEMP_getContext } from '~context/index';
+import { Action, ActionTypes, AllActions } from '~redux/index';
+import { putError, takeFrom, routeRedirect } from '~utils/saga/effects';
+import {
+  createTransaction,
+  createTransactionChannels,
+  getTxChannel,
+} from '../../../core/sagas';
+import { ipfsUpload } from '../../../core/sagas/ipfs';
+import {
+  transactionReady,
+  transactionPending,
+  transactionAddParams,
+} from '../../../core/actionCreators';
+
+function* managePermissionsMotion({
+  payload: {
+    colonyAddress,
+    domainId,
+    userAddress,
+    roles,
+    colonyName,
+    annotationMessage,
+  },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_MOTION_USER_ROLES_SET>) {
+  let txChannel;
+  try {
+    /*
+     * Validate the required values
+     */
+    if (!userAddress) {
+      throw new Error('User address not set for setUserRole transaction');
+    }
+
+    if (!domainId) {
+      throw new Error('Domain id not set for setUserRole transaction');
+    }
+
+    if (!roles) {
+      throw new Error('Roles not set for setUserRole transaction');
+    }
+
+    const context = TEMP_getContext(ContextModule.ColonyManager);
+    const colonyClient = yield context.getClient(
+      ClientType.ColonyClient,
+      colonyAddress,
+    );
+
+    const isRootDomain = domainId === ROOT_DOMAIN_ID;
+
+    const [permissionDomainId, childSkillIndex] = yield call(
+      getPermissionProofs,
+      colonyClient,
+      domainId,
+      isRootDomain ? ColonyRole.Root : ColonyRole.Architecture,
+    );
+
+    const { skillId } = yield call(
+      [colonyClient, colonyClient.getDomain],
+      ROOT_DOMAIN_ID,
+    );
+
+    const { key, value, branchMask, siblings } = yield call(
+      colonyClient.getReputation,
+      skillId,
+      AddressZero,
+    );
+
+    txChannel = yield call(getTxChannel, metaId);
+
+    // setup batch ids and channels
+    const batchKey = 'createRootMotion';
+
+    const {
+      createRootMotion,
+      annotateSetUserRolesMotion,
+    } = yield createTransactionChannels(metaId, [
+      'createRootMotion',
+      'annotateSetUserRolesMotion',
+    ]);
+
+    const roleArray = Object.values(roles).reverse();
+    roleArray.splice(2, 0, false);
+
+    let roleBitmask = '';
+
+    roleArray.forEach((role) => {
+      roleBitmask += role ? '1' : '0';
+    });
+
+    const hexString = hexlify(parseInt(roleBitmask, 2));
+    const zeroPadHexString = hexZeroPad(hexString, 32);
+
+    // eslint-disable-next-line max-len
+    const encodedAction = colonyClient.interface.functions.setUserRoles.encode([
+      permissionDomainId,
+      childSkillIndex,
+      userAddress,
+      domainId,
+      zeroPadHexString,
+    ]);
+
+    // create transactions
+    yield fork(createTransaction, createRootMotion.id, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createRootMotion',
```

Comment:
This is a rootMotion, when it is permission management in the root domain. I was not doing any domain motions so it didn't occur to me  but you have a good point that for now if it's not a root domain for the permissions there shouldn't be a motion trigger.
I can add this check or we can leave it like this for now since we will at some point implement domain motions - whichever you prefer.